ArjunaJTA/jta/classes/com/arjuna/ats/internal/jta/resources/arjunacore/XAResourceRecord.java

```diff
@@ -966,13 +966,17 @@ public boolean restore_state(InputObjectState os, int t)
 
 	                    // Give it a go ourselves
 	                    if (!deserialized) {
-	                        _theXAResource = (XAResource) o.readObject();
+	                        try {
+	                            _theXAResource = (XAResource) o.readObject();
+	                            if (jtaLogger.logger.isTraceEnabled()) {
+	                                jtaLogger.logger.trace("XAResourceRecord.restore_state - XAResource de-serialized");
+	                            }
+	                        } catch (ClassNotFoundException e) {
+	                            jtaLogger.i18NLogger.warn_resources_arjunacore_classnotfound(className);
+	                            _theXAResource = getNewXAResource();
```

Comment:
You need to check for if (_theXAResource == null) just like we do below in the case where we do not have a serialized resource (in the else on line 990). But rather than duplicate code you should just remove the else arm on line 990.